### PR TITLE
feat: use delete_message_seconds for Guild/Member.ban

### LINKED ISSUE #800: Use `delete_message_seconds` for `Guild/Member.ban`
### Summary

discord/discord-api-docs#5219

### What is the feature request for?

The core library

### The Problem

The new way to set a duration is with seconds not days, this will likely be removed in API V11.

*`reason` being removed means nothing to us as since API V10, all routes use the `X-Audit-Log-Reason` header*

### The Ideal Solution

`seconds` refers to `delete_message_seconds`, same with `days`
- Expose `seconds` to the user
- deprecate `days` but translate the given value to `seconds` (so no breaking changes)
- raise a `ValueError` if both provided (like `embed` and `embeds` are both mutually exclusive)

### The Current Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by teaishealthy**
## Summary

Resolves #800

## Checklist

- [ ] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
  - [x] I have run `task pyright` and fixed the relevant issues.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/guild.py [hunk 1]
```diff
@@ -2885,15 +2887,22 @@ async def ban(
...
 
+        .. warning::
+            delete_message_days is deprecated and will be removed in a future version.
+            Use delete_message_seconds instead.
+
         Parameters
         -----------
         user: :class:`abc.Snowflake`
             The user to ban from their guild.
-        delete_message_days: :class:`int`
```
Comment (EmreTech) [inline nextcord/guild.py:2892]: Why was this moved down below? Seems unnecessary TBH

File: nextcord/guild.py [hunk 2]
```diff
@@ -2874,7 +2875,8 @@ async def ban(
         user: Snowflake,
         *,
         reason: Optional[str] = None,
-        delete_message_days: Literal[0, 1, 2, 3, 4, 5, 6, 7] = 1,
+        delete_message_seconds: Optional[int] = None,
```
Comment (EmreTech) [inline nextcord/guild.py:2878]: This type hint should be set to `int` and its default should be set to `0` to match the HTTP route implementation

File: nextcord/guild.py [hunk 3]
```diff
@@ -2902,7 +2911,25 @@ async def ban(
...
+        if delete_message_days is not None:
+            warnings.warn(
+                DeprecationWarning(
+                    "delete_message_days is deprecated, use delete_message_seconds instead."
+                )
+            )
+            delete_message_seconds = delete_message_days * 24 * 60 * 60
+
+        await self._state.http.ban(
+            user.id, self.id, cast(int, delete_message_seconds), reason=reason
```
Comment (EmreTech) [inline nextcord/guild.py:2931]: Since casting an object doesn't actually do anything to the object itself at runtime, this is pretty unsafe if `delete_message_seconds` isn't passed into this method
First of all, it violates the type hints of the `ban` method and second of all I don't think Discord can handle `null` for that attribute

File: nextcord/member.py [hunk 4]
```diff
@@ -648,14 +648,20 @@ def communication_disabled_until(self) -> Optional[datetime.datetime]:
     async def ban(
         self,
         *,
-        delete_message_days: Literal[0, 1, 2, 3, 4, 5, 6, 7] = 1,
+        delete_message_seconds: Optional[int] = None,
```
Comment (EmreTech) [inline nextcord/member.py:651]: The type hint and defaults should also be changed here

**Review by EmreTech — CHANGES_REQUESTED** [review]

File: nextcord/guild.py (hunk 3 above)
Comment (EmreTech) [reply nextcord/guild.py:2931]: Resolved via a Discord conversation starting at this message: https://canary.discord.com/channels/881118111967883295/881124672056467456/1018006566097666098

File: nextcord/guild.py (hunk 2 above)
Comment (EmreTech) [reply nextcord/guild.py:2878]: https://github.com/nextcord/nextcord/pull/813#discussion_r967581183

File: nextcord/member.py (hunk 4 above)
Comment (EmreTech) [reply nextcord/member.py:651]: https://github.com/nextcord/nextcord/pull/813#discussion_r967581183

File: nextcord/guild.py [hunk 5]
```diff
@@ -2902,7 +2911,25 @@ async def ban(
         HTTPException
             Banning failed.
         """
-        await self._state.http.ban(user.id, self.id, delete_message_days, reason=reason)
+        if delete_message_days is not None and delete_message_seconds is not None:
+            raise InvalidArgument(
+                "Cannot pass both delete_message_days and delete_message_seconds."
+            )
+        elif delete_message_days is None and delete_message_seconds is None:
+            # Default to one day
```
Comment (Skelmis) [inline nextcord/guild.py:2919]: Can we document this default in the docstring?

**Review by EmreTech — APPROVED** [review]
LGTM, haven't tested

**Review by Skelmis — APPROVED** [review]
Untested, lgtm

File: nextcord/http.py [hunk 6]
```diff
@@ -879,12 +879,12 @@ def ban(
         self,
         user_id: Snowflake,
         guild_id: Snowflake,
-        delete_message_days: int = 1,
+        delete_message_seconds: int = 0,
```
Comment (DenverCoder1) [inline nextcord/http.py:882]: Maybe this should be the number of seconds in a day in case it is called directly. If it isn't ever called without it, there wouldn't be a need for a default at all.
```suggestion
        delete_message_seconds: int = 86400,
```

File: nextcord/guild.py [hunk 7]
```diff
@@ -2902,7 +2913,25 @@ async def ban(
...
+            warnings.warn(
+                DeprecationWarning(
+                    "delete_message_days is deprecated, use delete_message_seconds instead."
+                )
+            )
+            delete_message_seconds = delete_message_days * 24 * 60 * 60
+
+        await self._state.http.ban(
+            user.id, self.id, cast(int, delete_message_seconds), reason=reason
+        )
```
Comment (DenverCoder1) [inline nextcord/guild.py:2934]: You don't need to use `cast` if you have a condition that checks if `delete_message_seconds` is `None`.

```suggestion
        elif delete_message_days is not None:
            warnings.warn(
                DeprecationWarning(
                    "delete_message_days is deprecated, use delete_message_seconds instead."
                )
            )
            delete_message_seconds = delete_message_days * 24 * 60 * 60
        elif delete_message_seconds is None:
            # Default to one day
            delete_message_seconds = 24 * 60 * 60

        await self._state.http.ban(user.id, self.id, delete_message_seconds, reason=reason)
```

File: nextcord/guild.py [hunk 8]
```diff
@@ -43,6 +43,7 @@
     Set,
     Tuple,
     Union,
+    cast,
```
Comment (DenverCoder1) [inline nextcord/guild.py:46]: See above (cast is not needed)

```suggestion
```

**Review by DenverCoder1 — APPROVED** [review]

**Review by ooliver1 — APPROVED** [review]
